fedmsg/commands/check.py

```diff
@@ -0,0 +1,101 @@
+# This file is part of fedmsg.
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# fedmsg is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# fedmsg is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with fedmsg; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Authors: Jeremy Cline <jeremy@jcline.org>
+"""
+The ``fedmsg-check`` command can be used to see what fedmsg consumers and
+producers are currently active. It requires that the ``moksha.monitoring.socket``
+is defined in the fedmsg configuration.
+"""
+from gettext import gettext as _
+import json
+
+import click
+import zmq
+
+from fedmsg.config import load_config
+
+
+TIMEOUT_HELP = _("The number of seconds to wait for a heartbeat from the fedmsg hub."
+                 " The default is 10 seconds.")
+CONSUMER_HELP = _("The name of the consumer to check the status of.")
+PRODUCER_HELP = _("The name of the producer to check the status of.")
+
+
+@click.command()
+@click.option('--timeout', default=10, type=click.INT, help=TIMEOUT_HELP)
+@click.option('--consumer', '-c', multiple=True, help=CONSUMER_HELP)
+@click.option('--producer', '-p', multiple=True, help=PRODUCER_HELP)
+def check(timeout, consumer=None, producer=None):
+    """This command is used to check the status of consumers and producers.
+
+    If no consumers and producers are provided, the status of all consumers and producers
+    is printed.
+
+    Args:
+        timeout (int): The length of time in seconds to wait for a ZMQ message
+            on the monitor socket.
+        consumer (tuple): A list of consumer names to check the status of.
+        producer (tuple): A list of producer names to check the status of.
```

Comment:
I recommend against documenting the args like this on click commands - if the user types ```--help``` on this command, this entire docblock will be displayed and won't be particularly helpful or useful to the user. It will also give duplicate info since the help will also already tell them about the flags.

See https://github.com/fedora-infra/bodhi/issues/1457